crates/ledger/src/rewards.rs

```diff
@@ -151,58 +151,55 @@ impl StakeDistribution {
     /// Clompute a new stake distribution snapshot using data available in the `Store`.
     ///
     /// Invariant: The given store is expected to be a snapshot taken at the end of an epoch.
-    pub fn new<E>(db: &impl Store<Error = E>) -> Result<Self, E> {
-        let mut accounts = BTreeMap::new();
-        db.with_accounts(|rows| {
-            for (credential, row) in rows {
-                if let Some(account) = row.borrow() {
-                    if let Some(pool) = account.delegatee {
-                        accounts.insert(
-                            credential,
-                            AccountState {
-                                pool,
-                                lovelace: account.rewards,
-                            },
-                        );
-                    }
-                }
-            }
-        })?;
-
-        db.with_utxo(|rows| {
-            for (_, row) in rows {
-                if let Some(output) = row.borrow() {
-                    if let Some(credential) = output_stake_credential(output) {
-                        let value = output_lovelace(output);
-                        accounts
-                            .entry(credential)
-                            .and_modify(|account| account.lovelace += value);
-                    }
-                }
-            }
-        })?;
-
-        let mut pools: BTreeMap<PoolId, PoolState> = BTreeMap::new();
-        db.with_pools(|rows| {
-            for (pool, row) in rows {
-                if let Some(row) = row.borrow() {
-                    pools.insert(
+    pub fn new(db: &impl Snapshot) -> Result<Self, StoreError> {
+        // TODO: Avoid creating this intermediate map, and directly create the keys/scripts ones.
+        // Then, when looking for _active accounts_ delegated to pools, we can prune those not
+        // referenced anywhere.
+        let mut accounts = db
+            .iter_accounts()?
+            .filter_map(|(credential, account)| {
+                account.delegatee.map(|pool| (pool, credential, account))
```

Comment:
```suggestion
                account.delegatee.map(|pool| (credential, AccountState {
                      pool,
                      lovelace: account.rewards,
                })
```

Reply:
The first `map` is nested inside the `filter_map`, those are not 2 maps in a row.